pkg/solana/txm/txm.go

```diff
@@ -378,135 +453,203 @@ func (txm *Txm) confirm() {
 		case <-ctx.Done():
 			return
 		case <-tick:
-			// get list of tx signatures to confirm
+			client, err := txm.client.Get()
+			// Get list of transaction signatures to confirm
+			// If no signatures to confirm, we can break loop.
 			sigs := txm.txs.ListAll()
-
-			// exit switch if not txs to confirm
 			if len(sigs) == 0 {
 				break
 			}
 
-			// get client
-			client, err := txm.client.Get()
 			if err != nil {
-				txm.lggr.Errorw("failed to get client in soltxm.confirm", "error", err)
-				break // exit switch
+				txm.lggr.Errorw("failed to get client in txm.confirm", "error", err)
+				return
 			}
-
-			// batch sigs no more than MaxSigsToConfirm each
-			sigsBatch, err := utils.BatchSplit(sigs, MaxSigsToConfirm)
-			if err != nil { // this should never happen
-				txm.lggr.Fatalw("failed to batch signatures", "error", err)
-				break // exit switch
+			if txm.cfg.TxExpirationRebroadcast() {
+				txm.rebroadcastExpiredTxs(ctx, client)
 			}
```

Comment:
We may want to do this after we process confirmation in case a transaction gets processed between the last check and this one close to its expiration slot. We could potentially rebroadcast a transaction we think is expired even though it was included. But this does remind me that we probably need to be careful with `BlockhashNotFound` errors in the confirm loop so we're not marking expired tx as errored if `TxExpirationRebroadcast` is enabled